### ISSUE TITLE
Resources with invalid permissions are getting cached

### ISSUE THREAD

**Issue by forbushbl** [cloudfoundry/cloud_controller_ng#1705]
## Issue

When a user attempts to `cf push` an app that contains a file with 444 permissions, it succeeds the first time but fails on future attempts. The cause of this problem is that files with invalid permissions (i.e. file mode < 0600) are getting cached by CC even though clients can't reference those cached files.

## Context

When a file with invalid permissions is uploaded to CC via `PUT /v2/apps/:guid/bits` it gets cached. Later on when a client tries to reference the cached file in a subsequent call to `PUT /v2/apps/:guid/bits` by including it in the `resources` field they see the following error:

    Job (446db953-aef5-4c24-bf75-ad623b94ef8a) failed: The resource file mode is invalid: File mode '444' with path 'cacheable.stuff' is invalid. Minimum file mode is '0600'

The [API documentation](https://apidocs.cloudfoundry.org/13.5.0/apps/uploads_the_bits_for_an_app.html) states that files must have at least 0600. 

## Steps to Reproduce

**Prerequisites:**
* Resource caching must be enabled on your CC

**Steps:**
1. Create a file which is larger than the configured `cc.resource_pool.minimum_size` and change it's permissions to 444
    ```
    head -c 65537 </dev/urandom > cacheable.stuff
    chmod 444 cacheable.stuff
    ```
2. Create a file which is smaller than the configured `cc.resource_pool.minimum_size`
    ```
    echo "<html><body>Hello world</body></html>" > index.html
    ```
    Your directory should look something like this
    ```
    $ ls -l
    -r--r--r--  1 user  staff  65537 Jun 26 14:04 cacheable.stuff
    -rw-r--r--  1 user  staff    119 Jun 26 11:31 index.html
    ```
4. Do a `cf push`
    ```
    $ cf push myapp
    ...
    It succeeds!
    ```
5. Do a `cf push` again
    ```
    $ cf push myapp
    ...
    Packaging files to upload...
    Uploading files...
    232 B / 232 B [=========================================================================================================================================================] 100.00% 1s

    Waiting for API to complete processing files...
    Job (f0c0e4cb-9628-4878-86cc-620d4e424510) failed: The resource file mode is invalid: File mode '444' with path 'cacheable.stuff' is invalid. Minimum file mode is '0600'
    FAILED
    ```

## Expected result

I think the best solution would be to simply avoid cacheing any files are don't meet the minimum permissions of 0600.

## Current result

The problem with the CLI goes something like this.

**cf push (first time)**
1. Check /v2/resource_match to see if files are already cached. They aren't.
2. Upload files and CC caches them, regardless of file permissions.

**cf push (second time)**
1. Check /v2/resource_match to see if files are already cached. They are!
2. Upload non-cached files and references to cached files
3. CC complains that the client is trying to access a cached file with < 0600 permissions. 😞

**Comment by forbushbl** [comment]
See the following conversation for additional context: https://cloudfoundry.slack.com/archives/C07C04W4Q/p1593045107159800

**Comment by Gerg** [comment]
The resolution for this issue is included in [capi-release 1.102](https://github.com/cloudfoundry/capi-release/releases/tag/1.102.0). Thanks for bringing this issue to our attention!
